addons/osfstorage/models.py

```diff
@@ -337,26 +337,13 @@ class OsfStorageFolder(OsfStorageFileNode, Folder):
 
     @property
     def is_checked_out(self):
-        try:
-            if self.checkout:
-                return True
-        except AttributeError:
-            return False
-        # TODO this should be one query
-        for child in self.children.all():
-            try:
-                if child.is_checked_out:
-                    return True
-            except AttributeError:
-                pass
-        return False
+        return self.children.filter(checkout__isnull=False).exists()
```

Comment:
I looked this over and decided to add two more tests to my blocking ticket (in CR). I'm pretty sure this is going to allow deep nested checked-out files to be moveable to another component. Sorry for missing that in my tests the first time around.